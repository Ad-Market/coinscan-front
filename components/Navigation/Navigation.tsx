import { useRouter } from 'next/router';
import Divider from '@mui/material/Divider';
import Button from '@mui/material/Button';
import Stack from '@mui/material/Stack';
import Link from '../Link';

const Navigation = () => {
  const route = useRouter();
  return (
    <Stack
      direction="row"
      divider={<Divider orientation="vertical" flexItem />}
      spacing={2}
    >
      <Button
        component={Link}
        href="/"
        variant={
          route.pathname === '/' || route.pathname.includes('/wallet')
            ? 'outlined'
            : 'text'
        }
      >
        Wallets
      </Button>

      <Button
        component={Link}
        href="/transactions"
        variant={route.pathname.includes('/transactions') ? 'outlined' : 'text'}
      >
        Transactions
      </Button>
    </Stack>
  );
};
export default Navigation;
