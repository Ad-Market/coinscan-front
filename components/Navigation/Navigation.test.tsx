import { render } from '@testing-library/react';
import Navigation from './Navigation';

jest.mock('next/router', () => require('next-router-mock'));

test('Navigation renders', () => {
  const { container } = render(<Navigation />);
  expect(container).toMatchSnapshot('navigation rendered');
});
