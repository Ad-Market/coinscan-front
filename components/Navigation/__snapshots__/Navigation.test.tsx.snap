// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders: navigation rendered 1`] = `
<div>
  <div
    class="css-e53awj-MuiStack-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-mph559-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
      href="/"
      tabindex="0"
    >
      Wallets
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
    />
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-mph559-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
      href="/transactions"
      tabindex="0"
    >
      Transactions
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
